http/oauth2/oauth2_test.go

```diff
@@ -80,6 +81,143 @@ func TestHandlerIntrospectError(t *testing.T) {
+			r.Header.Add("Authorization", "Bearer bearer")
+			authorize, b := tC.auth.Authorize(r, w)
+
+			if !b || w.Code != http.StatusOK || authorize == nil {
+				t.Errorf("Expected succesfull Authentication, but was not succesfull with code %d and body %v", w.Code, w.Body)
+				return
+			}
+			to, _ := security.GetTokenFromContext(authorize)
+			tok, ok := to.(*token)
+
+			if !ok || tok.value != "bearer" || tok.scope != Scope("ABC DHHG kjdk") || tok.clientID != "ClientId" || tok.userID != "UserId" {
+				t.Errorf("Expected %v but got %v", tC.auth.introspection.(*tokenIntrospectedSuccessful).response, tok)
+			}
+		})
+	}
+}
+
+func TestAuthenticationSuccessScopeError(t *testing.T) {
+	auth := NewAuthenticator(&tokenIntrospectedSuccessful{&IntrospectResponse{
+		Active:   true,
+		Scope:    "ABC DEF DFE",
+		ClientID: "ClientId",
+		UserID:   "UserId",
+	}}, &Config{}).WithScope("DE")
+
+	w := httptest.NewRecorder()
+	r := httptest.NewRequest("GET", "/", nil)
+	r.Header.Add("Authorization", "Bearer bearer")
+
+	_, b := auth.Authorize(r, w)
+
+	resp := w.Result()
+	body, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if b {
+		t.Errorf("Expected error in Authentication, but was succesfull with code %d and body %v", resp.StatusCode, string(body))
+	}
+	if got, ex := w.Code, http.StatusForbidden; got != ex {
+		t.Errorf("Expected status code %d, got %d", ex, got)
+	}
+	if got, ex := string(body), "Forbidden - requires scope \"DE\"\n"; got != ex {
+		t.Errorf("Expected status code %q, got %q", ex, got)
+	}
+}
+
+func TestAuthenticationWithErrors(t *testing.T) {
+	testCases := []struct {
+		desc         string
+		returnedErr  error
+		expectedCode int
+		expectedBody string
+	}{
+		{
+			desc:         "token introspecter returns ErrBadUpstreamResponse",
+			returnedErr:  ErrBadUpstreamResponse,
+			expectedCode: 502,
+			expectedBody: "bad upstream response when introspecting token\n",
+		},
+		{
+			desc:         "token introspecter returns ErrUpstreamConnection",
+			returnedErr:  ErrUpstreamConnection,
+			expectedCode: 502,
+			expectedBody: "problem connecting to the introspection endpoint\n",
+		},
+		{
+			desc:         "token introspecter returns ErrInvalidToken",
+			returnedErr:  ErrInvalidToken,
+			expectedCode: 401,
+			expectedBody: "user token is invalid\n",
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			auth := NewAuthenticator(&tokenInspectorWithError{returnedErr: tC.returnedErr}, &Config{})
+			w := httptest.NewRecorder()
+			r := httptest.NewRequest("GET", "/", nil)
+			r.Header.Add("Authorization", "Bearer bearer")
+			_, b := auth.Authorize(r, w)
+
+			resp := w.Result()
+			body, err := ioutil.ReadAll(resp.Body)
+			defer resp.Body.Close()
+			if err != nil {
+				t.Fatal(err)
+			}
+			if b {
+				t.Errorf("Expected error in Authentication, but was succesfull with code %d and body %v", resp.StatusCode, string(body))
```

Comment:
No code and body if it was successful. Also typo:
```suggestion
				t.Errorf("Expected error in authentication, but was succesful with code %d and body %v", resp.StatusCode, string(body))
```